### PR TITLE
Update username algo

### PR DESCRIPTION
**Pull request by asood123**
- Cleans up username algorithm
- introduces `suggestedUsername` field on User, as a way to pass github usernames (or other ones in the future
- looks at group slug table for conflicts.

### CODE REVIEW

File: server/lib/userlib.js [hunk 1]
```diff
@@ -113,5 +114,55 @@ export default {
...
     });
-  }
+  },
+
+  /*
+   * If there is a username suggested, we'll check that it's valid or increase it's count
+   * Otherwise, we'll suggest something.
+   */
+
+  checkOrSuggestUsername(user) {
```
Comment (xdamman) [inline server/lib/userlib.js]: It's a weird method name. Check + Or don't go well together (what are you checking before getting to the OR part?). I'd prefer in this case simply createUsername()

File: server/lib/userlib.js [hunk 2]
```diff
@@ -113,5 +114,55 @@ export default {
...
+    // In theory, this should never happen because we already have an email
+    // TODO: add a random username to make sure that every user has a username
+    if (potentialUserNames.length === 0) {
+      console.error(`No potential username found for user: ${user.email}`);
+      return Promise.resolve()
+    }
+
+    // fetch any matching usernames or slugs for the top choice in the list above
+    return sequelize.query(`
+        SELECT username as username FROM "Users" where username like '${potentialUserNames[0]}%'
```
Comment (xdamman) [inline server/lib/userlib.js]: Oh so you too are getting into raw queries? He he :-)

Comment (asood123) [inline server/lib/userlib.js]: no way to do UNION ALL in sequelize that I can find :/

**Comment by asood123** [comment]
fixed. merging now.

**Pull request merged**